src/sentry/event_manager.py

```diff
@@ -1829,11 +1829,15 @@ def _create_group(project: Project, event: Event, **group_creation_kwargs: Any)
     # add sdk tag to metadata
     group_data.setdefault("metadata", {}).update(sdk_metadata_from_event(event))
 
-    # add severity to metadata for alert filtering
-    severity = _get_severity_metadata_for_group(event)
-    group_data["metadata"].update(severity)
+    # Add severity to metadata for alert filtering for errors events.
+    # We can skip this if the group type is explicitly NOT an error.
+    group_type = group_creation_kwargs.get("type", None)
+    if not group_type or group_type == ErrorGroupType.type_id:
+        severity = _get_severity_metadata_for_group(event)
+        group_data["metadata"].update(severity)
 
     if features.has("projects:issue-priority", project, actor=None):
+        # the kwargs only include priority for non-error issue platform events, which takes precedence.
         priority = group_creation_kwargs.get("priority", None)
         if priority is None:
             priority = _get_priority_for_group(severity, group_creation_kwargs)
```

Comment:
It looks like this function's type signature should be changed so that `severity` can be `None`?

https://github.com/getsentry/sentry/blob/e467f06a7fa83ad79c3f201a966f7e72df066f93/src/sentry/event_manager.py#L2182

Reply:
Ah didn't catch that it was unbounded - fixed it!